crates/bevy_pbr/src/wireframe.rs

```diff
@@ -38,138 +29,169 @@ impl Plugin for WireframePlugin {
 
         app.register_type::<WireframeConfig>()
             .init_resource::<WireframeConfig>()
-            .add_plugin(ExtractResourcePlugin::<WireframeConfig>::default());
-
-        if let Ok(render_app) = app.get_sub_app_mut(RenderApp) {
-            render_app
-                .add_render_command::<Opaque3d, DrawWireframes>()
-                .init_resource::<WireframePipeline>()
-                .init_resource::<SpecializedMeshPipelines<WireframePipeline>>()
-                .add_system_to_stage(RenderStage::Extract, extract_wireframes)
-                .add_system_to_stage(RenderStage::Queue, queue_wireframes);
-        }
-    }
-}
-
-fn extract_wireframes(mut commands: Commands, query: Extract<Query<Entity, With<Wireframe>>>) {
-    for entity in query.iter() {
-        commands.get_or_spawn(entity).insert(Wireframe);
+            .add_plugin(ExtractResourcePlugin::<WireframeConfig>::default())
+            .add_plugin(MaterialPlugin::<WireframeMaterial>::default())
+            .add_system(apply_global)
+            .add_system(apply_material)
+            .add_system(wireframe_color_changed)
+            .add_system(global_color_changed);
     }
 }
 
-/// Controls whether an entity should rendered in wireframe-mode if the [`WireframePlugin`] is enabled
+/// Toggles wireframe rendering for any entity it is attached to.
+///
+/// This requires the [`WireframePlugin`] to be enabled.
 #[derive(Component, Debug, Clone, Default, Reflect)]
 #[reflect(Component, Default)]
 pub struct Wireframe;
 
+/// Sets the color of the [`Wireframe`] of the entity it is attached to.
+///
+/// This overrides the [`WireframeConfig::color`].
```

Comment:
We should explain the behavior if this component is absent, but [`Wireframe`] is missing.